### PR TITLE
TN Aggregate Ingest

### PR DESCRIPTION
**Pull request by ohinds**

### CODE REVIEW

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+
+_MANUAL_FACILITY_TO_COUNTY_MAP = {
+    'Johnson City (F)': 'Washington',
+    'Kingsport City': 'Sullivan',
+}
+
+
+def parse(filename: str) -> Dict:
+
```
Comment (jessex) [inline recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Extra newline.

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+from recidiviz.persistence.database.schema import TnFacilityAggregate
+
+
+_MANUAL_FACILITY_TO_COUNTY_MAP = {
+    'Johnson City (F)': 'Washington',
+    'Kingsport City': 'Sullivan',
+}
+
+
+def parse(filename: str) -> Dict:
```
Comment (zacharylawrence) [inline recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Could you pass the generic type information of the keys/values into the `Dict`?

Like: `Dict[DeclarativeMeta, pd.DataFrame]`

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        r'Pre- trial Felony': 'pretrial_felony_population',
+        r'Pre- trial Misd\.': 'pretrial_misdemeanor_population',
+        r'Total Jail Pop\.': 'total_jail_population',
+        r'Total Beds\*\*': 'total_beds',
+    }
+
+    df = aggregate_ingest_utils.rename_columns_and_select(
+        df, rename, use_regex=True)
+
+    df = df.dropna(how='all')
```
Comment (zacharylawrence) [inline recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Could you add a quick comment about how some rows are parsed into 2 lines but none of the columns we care about are split over the two lines (& that's why we need to do this)?

File: recidiviz/persistence/database/schema.py [hunk 4]
```diff
@@ -928,3 +928,24 @@ class PaFacilityPopAggregate(Base, _AggregateTableMixin):
     work_release_adp = Column(Integer)
     admissions = Column(Integer)
     discharge = Column(Integer)
+
+
+class TnFacilityAggregate(Base, _AggregateTableMixin):
+    """TN state-provided aggregate population statistics."""
+    __tablename__ = 'tn_facility_pop_aggregate'
```
Comment (zacharylawrence) [inline recidiviz/persistence/database/schema.py]: Could you update this tablename to match the class name? `tn_facility_aggregate`

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py (hunk 2 above)
Comment (zacharylawrence) [inline recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: This is a super nit so feel free to ignore it: In the other aggregate parses we separate the "parse what's on the pdf" logic from the "add stuff to the parsed data" logic. I think it's minor, but you can see in [ga_aggregate_ingest.py](https://github.com/Recidiviz/recidiviz-data/blob/2cbd41f67e5941121f0d29eddb181389e26cc578/recidiviz/ingest/aggregate/regions/ga/ga_aggregate_ingest.py#L36) how we have a main `parse` method which adds fips/report_date/report_granularity and knows about the schema vs the `_parse_table` which just operates on the data in the pdf and nothing else.

If you like it more as is then I'm fine with it. Thoughts?

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+    df['federal_and_other_population'] = df[
```
Comment (zacharylawrence) [inline recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Nit: Could you add a quick comment here about how tabula parses 2 columns into 1 so we have to manually split?

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    'Johnson City (F)': 'Washington',
+    'Kingsport City': 'Sullivan',
+}
+
+
+def parse(filename: str) -> Dict:
+
+    table = tabula.read_pdf(filename, pages=[2, 3, 4], multiple_tables=True)
+
+    formatted_dfs = []
```
[the remark this answers is not in the document]
Comment (ohinds) [reply recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Nice, done.

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py (hunk 2 above)
Comment (ohinds) [reply recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Done.

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py (hunk 3 above)
Comment (ohinds) [reply recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Done.

File: recidiviz/persistence/database/schema.py (hunk 4 above)
Comment (ohinds) [reply recidiviz/persistence/database/schema.py]: Whoops!

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py (hunk 2 above)
Comment (ohinds) [reply recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Good idea. Done.

File: recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py (hunk 1 above)
Comment (jessex) [reply recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: I _think_ you're misinterpreting that. I think it's saying two lines between the end of a top-level definition and the start of a new one, not being, say, the signature of a top-level method and the first line of its implementation. So two lines between the end of `_MANUAL_FACILITY_TO_COUNTY_MAP` and the start of `parse`, not an extra line between the start of `parse` and `table = parse_table(filename)`. Does that make sense?

Comment (ohinds) [reply recidiviz/ingest/aggregate/regions/tn/tn_aggregate_ingest.py]: Haha, I just missed which line you were calling extra. Removed.

**Pull request merged**